gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -414,30 +415,49 @@ public void run() {
       }
     }
 
+    /**
+     * Cancels the dag and sends a cancellation tracking event.
```

Comment:
Job level cancellation tracking event or Flow level cancellation tracking event? It looks as though you are emitting job level cancellation event. However, I am not convinced it is right. The Job cancel timer should come from the executor. Else, there is a chance that the job cancel event emitted from cancelDag() may be overwritten by job running timer(s) from the executor leading to incorrect job status. On the other hand, a flow level timer events will be only emitted from GaaS and will correctly reflect the overall status of the flow.

Reply:
sending a flow level event now, but changes for processing it will be done in next PR